src/atomate2/ase/schemas.py

```diff
@@ -0,0 +1,479 @@
+
+            trajectory = getattr(PmgTrajectory, traj_method)(
+                [input_config],
+                frame_properties=[trajectory.frame_properties[0]],
+                constant_lattice=False,
+            )
+            output_config = input_config
+        else:
+            output_config = result["final_ionic_config"]
+
+        final_energy = trajectory.frame_properties[-1]["energy"]
+        final_energy_per_atom = final_energy / len(input_config)
+        final_forces = trajectory.frame_properties[-1]["forces"]
+        final_stress = trajectory.frame_properties[-1].get("stress")
+
+        ionic_steps = []
+        for idx in range(n_steps):
+            _ionic_step_data = {
+                key: trajectory.frame_properties[idx].get(key)
+                if key in ionic_step_data
+                else None
+                for key in ("energy", "forces", "stress")
+            }
+
+            current_config = (
+                trajectory[idx] if "ionic_config" in ionic_step_data else None
+            )
+
+            # include "magmoms" in :obj:`ionic_step` if the trajectory has "magmoms"
+            if "magmoms" in trajectory.frame_properties[idx]:
+                _ionic_step_data.update(
+                    {
+                        "magmoms": trajectory.frame_properties[idx]["magmoms"]
+                        if "magmoms" in ionic_step_data
+                        else None
+                    }
+                )
+
+            ionic_step = IonicStep(
+                ionic_configuration=current_config,
+                **_ionic_step_data,
+            )
+
+            ionic_steps.append(ionic_step)
+
+        objects: dict[AseObject, Any] = {}
+        if store_trajectory != StoreTrajectoryOption.NO:
+            # For VASP calculations, the PARTIAL trajectory option removes
+            # electronic step info. There is no equivalent for classical
+            # forcefields, so we just save the same info for FULL and
+            # PARTIAL options.
+            objects[AseObject.TRAJECTORY] = trajectory  # type: ignore[index]
+
+        output_doc = OutputDoc(
+            ionic_configuration=output_config,
+            energy=final_energy,
+            energy_per_atom=final_energy_per_atom,
+            forces=final_forces,
+            stress=final_stress,
+            ionic_steps=ionic_steps,
+            n_steps=n_steps,
+        )
+
+        return cls(
+            ionic_configuration=output_config,
+            input=input_doc,
+            output=output_doc,
+            ase_calculator_name=ase_calculator_name,
+            included_objects=list(objects.keys()),
+            objects=objects,
+            is_force_converged=result.get("is_force_converged"),
+            energy_downhill=result.get("energy_downhill"),
+            dir_name=result.get("dir_name"),
+            **task_document_kwargs,
+        )
+
+    def to_meta_task_doc(self) -> Union[AseStructureTaskDoc, AseMoleculeTaskDoc]:
```

Comment:
We need a public `from_ase_compatible_result` at least in the forcefield doc to prevent breaking changes. The `to_meta_task_doc` method is only used on the ase side, I'll try to clean that up a bit